test/srt/ops/test_repeat_interleave.py

```diff
@@ -0,0 +1,153 @@
+            tw = torch.randint(1, 17, (T, 1), dtype=torch.int32)
+            grid_thw = torch.cat([repeats, th, tw], dim=1)
+
+            # Save a copy to ensure functions do not modify inputs
+            grid_clone = grid_thw.clone()
+
+            out_torch = torch_ref_impl(grid_thw)
+            out_numpy = cpu_numpy_impl(grid_thw)
+
+            # Input should not be modified
+            assert torch.equal(
+                grid_thw, grid_clone
+            ), "Function modified input grid_thw!"
+
+            # Shapes must be the same
+            assert (
+                out_torch.shape == out_numpy.shape
+            ), f"Shape mismatch: torch={out_torch.shape}, numpy={out_numpy.shape}"
+
+            # Dtypes must be the same (should both be int32)
+            assert (
+                out_torch.dtype == out_numpy.dtype == torch.int32
+            ), f"dtype mismatch: torch={out_torch.dtype}, numpy={out_numpy.dtype}"
+
+            # Values must be exactly the same
+            if not torch.equal(out_torch.cpu(), out_numpy.cpu()):
+                diff_idx = (out_torch.cpu() != out_numpy.cpu()).nonzero(as_tuple=False)
+                idx0 = diff_idx[0].item()
+                raise AssertionError(
+                    f"Value mismatch, T={T}, first differing index={idx0}, "
+                    f"torch={out_torch[idx0].item()}, "
+                    f"numpy={out_numpy[idx0].item()}"
+                )
+
+    print("CPU correctness check: PASSED.")
+
+
+def main():
+    # Setting number of threads to reduce noise from thread scheduling;
+    # you can comment this out if you prefer default behavior.
+    torch.set_num_threads(1)
+
+    # --------------- Correctness check ---------------
+    check_correctness_cpu()
+    print("\nAll correctness checks passed. Starting benchmark...\n")
+
+    # --------------- Performance benchmark ---------------
+    # Typical scales:
+    # T = number of rows in grid_thw
+    # H, W only participate in multiplication
+    configs = [
+        (128, 8, 8),
+        (512, 8, 8),
+        (2048, 8, 8),
+        (8192, 8, 8),
+    ]
+
+    iters = 2000  # number of iterations per configuration
+
+    print("=== CPU benchmark ===")
+    for T, H, W in configs:
+        # Construct grid_thw: [T, 3]
+        # col0: repeat count
+        # col1, col2: multiplicative factors
+        grid_thw = torch.randint(1, 5, (T, 3), dtype=torch.int32)
+        grid_thw[:, 1] = H
+        grid_thw[:, 2] = W
+
+        t_torch, out_torch = benchmark_once(torch_ref_impl, grid_thw, iters=iters)
+        t_numpy, out_numpy = benchmark_once(cpu_numpy_impl, grid_thw, iters=iters)
+
+        # Additional safety check: results should match
+        same = torch.equal(out_torch.cpu(), out_numpy.cpu())
+
+        print(
+            f"[CPU] T={T:5d}, iters={iters:4d} | "
+            f"torch={t_torch*1e3:7.2f} ms, "
+            f"numpy={t_numpy*1e3:7.2f} ms, "
+            f"same={same}"
+        )
+
+
```

Comment:
Please refactor the test into a unit-test style, keeping it consistent with the other tests under the `srt` directory.